fix: styling and layout changes for dashboard, footer and product page

arslanashraf7 opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
#91 

#### What's this PR do?
- Fixes the styling/layout issues in the footer
- Fixes the layout issues on the dashboard
- Fixes the layout issues on the Product Detail Page
- Removed extra <p> tags from rich text in product detail page
- Removed xPRO's unused style from the footer which was copied over xPRO

Putting simply, these are the things from PDF:
1. All footer text to be `font-weight:400` & spacing Mentioned in PDF
2. For the dashboard page, adjusted the spacing mentioned in the PDF
3. For product page background is set to `white` and spacing adjusted as per PDF

**Note:** There is a pdf file attached to the associated issue which will help out identifying the required styling changes

#### How should this be manually tested?
- Verify it against the changes mentioned in the PDF attached to the ticket.
- Verify the nothing else has been affected by this PR

#### Where should the reviewer start?
- HomePage

#### Screenshots (if appropriate)
**Footer Desktop**
<img width="1440" alt="Screenshot 2021-08-17 at 5 00 17 PM" src="https://user-images.githubusercontent.com/34372316/129725857-bcfa595c-a5ad-47af-be51-a69d259ccbd8.png">

**Footer Mobile**
<img width="500" alt="Screenshot 2021-08-17 at 5 00 30 PM" src="https://user-images.githubusercontent.com/34372316/129725865-d35dcc03-1e12-41d9-9408-1c500dcab564.png">

**Dashboard Desktop**
<img width="1440" alt="Screenshot 2021-08-17 at 5 00 57 PM" src="https://user-images.githubusercontent.com/34372316/129725869-40e02922-13bc-44bc-b84c-ac000cf7d9eb.png">

**Dashboard Mobile**
<img width="500" alt="Screenshot 2021-08-17 at 5 01 07 PM" src="https://user-images.githubusercontent.com/34372316/129725872-d36a692b-38a3-4433-848f-a55628dc7e28.png">

**Product Page Desktop**
<img width="1425" alt="Screenshot 2021-08-17 at 5 28 24 PM" src="https://user-images.githubusercontent.com/34372316/129725875-1216abd3-9294-4bc3-9cb6-58dc26259cc9.png">

**Product Page Mobile**
<img width="499" alt="Screenshot 2021-08-17 at 5 30 30 PM" src="https://user-images.githubusercontent.com/34372316/129725878-81805a91-1970-4836-80a6-e74c497786ab.png">

gsidebo commented on static/scss/dashboard.scss:
```diff
@@ -37,4 +37,18 @@ $featureImgMobileWidth: 343px;
+}
+
+.courses-heading {
+  font-size: 40px;
```
This font-size should be applied to the general style in `common.scss` so all `h1`'s use it. There is no reason to have some `h1` elements at 40px and others at 42px/48px. Those values are close enough that you should always assume that the change should be applied to all elements like it. You can also try applying the mobile styles to the `common.scss` and see how the other `h1` elements on other pages look.

You can also remove the `courses-heading` class from that `h1` if the above suggestions work out

gsidebo reviewed (changes requested):
There are a couple other possible refactors, but this is good to merge after the `h1` thing is addressed

arslanashraf7 commented:
>This font-size should be applied to the general style in common.scss so all h1's use it. There is no reason to have some h1 elements at 40px and others at 42px/48px. Those values are close enough that you should always assume that the change should be applied to all elements like it. You can also try applying the mobile styles to the common.scss and see how the other h1 elements on other pages look.

I like this idea and I've updated the font size/ lint-height styles `common.scss` for `h1`. Instead of setting `40px` for all `h1` i took the liberty of setting `font-size/line-height` as `42px/52px` for desktops and `34px/38px` for mobile. The reason I tried to set `42px` instead of `40px` was to make sure that we could make the h1 style common keeping it close to the existing design and also being able to do the required styling that's part of this ticket.

Attaching some before/after screenshots below so that you can also take a look at the difference:

<img width="711" alt="Screenshot 2021-08-23 at 3 54 50 PM" src="https://user-images.githubusercontent.com/34372316/130438608-14c29cbe-db48-40dd-82fd-fa1ed3075638.png">
<img width="671" alt="Screenshot 2021-08-23 at 3 54 58 PM" src="https://user-images.githubusercontent.com/34372316/130438610-e73749cf-d197-4aeb-a3b4-4e874aaa1758.png">

<img width="646" alt="Screenshot 2021-08-23 at 3 55 11 PM" src="https://user-images.githubusercontent.com/34372316/130438615-34f488d3-b787-48b8-8ff3-b1b7fd9a24e2.png">
<img width="682" alt="Screenshot 2021-08-23 at 3 55 20 PM" src="https://user-images.githubusercontent.com/34372316/130438617-eca0f8be-f809-4e41-94f4-0ba3692d6e6d.png">

<img width="878" alt="Screenshot 2021-08-23 at 3 52 34 PM" src="https://user-images.githubusercontent.com/34372316/130438579-eb3e0d1f-67b8-433b-b651-14a57ee730b5.png">
<img width="857" alt="Screenshot 2021-08-23 at 3 52 26 PM" src="https://user-images.githubusercontent.com/34372316/130438572-10ba95d3-62fe-4311-a4e9-267608ce1015.png">

<img width="859" alt="Screenshot 2021-08-23 at 3 52 52 PM" src="https://user-images.githubusercontent.com/34372316/130438582-3c74d4ab-ea4f-4d4b-9c2c-c98c39f16200.png">
<img width="848" alt="Screenshot 2021-08-23 at 3 52 57 PM" src="https://user-images.githubusercontent.com/34372316/130438587-9bbf3a00-10cf-419a-86f9-888bd7def42d.png">

<img width="1218" alt="Screenshot 2021-08-23 at 3 53 18 PM" src="https://user-images.githubusercontent.com/34372316/130438595-c9b001a0-afda-4893-9b45-d37e73396e09.png">
<img width="1256" alt="Screenshot 2021-08-23 at 3 53 26 PM" src="https://user-images.githubusercontent.com/34372316/130438599-ea3f8f44-da92-42a6-8ab9-8802c37cccdd.png">

<img width="1115" alt="Screenshot 2021-08-23 at 3 53 45 PM" src="https://user-images.githubusercontent.com/34372316/130438605-dfca1eb2-be5a-42a4-a06a-b989436fd81e.png">
<img width="1312" alt="Screenshot 2021-08-23 at 3 53 40 PM" src="https://user-images.githubusercontent.com/34372316/130438603-7f727f5b-89e0-4c3d-bfd9-998ec56615d6.png">
<img width="905" alt="Screenshot 2021-08-23 at 4 30 19 PM" src="https://user-images.githubusercontent.com/34372316/130440254-8fad373a-03cc-42d4-b43a-e88fcf16a690.png">
<img width="906" alt="Screenshot 2021-08-23 at 4 30 11 PM" src="https://user-images.githubusercontent.com/34372316/130440280-a993c244-defc-4982-ab00-450588f75f93.png">


>You can also remove the courses-heading class from that h1 if the above suggestions work out

- I've removed the size and line height-related styling from the class but I've kept the class for spacing-related styles that in my opinion might change in different places e.g. specifically on the dashboard.

If you are fine with these changes we can merge them or if you think the design changes are too broad I will revert the last commit.


CC: @pdpinch (Tagging because this might introduce some design tweaks, not very major but wanted to let you know in any case)

pdpinch commented:
I'm checking with the designer why the course detail page has a larger H1 than the other account creation pages.

pdpinch commented:
InVision has been updated, https://projects.invisionapp.com/d/main?origin=v7#/projects/prototypes/21535972?inbox=true&inboxProjectID=21535972&inboxConversationID=144914028

In any case, the h1 on the course detail should be the same size as all the rest.

gsidebo reviewed (approved):
Looks good 👍 . You can still get rid of the `courses-heading` class as I suggested before, and just style the `h1` directly in `dashboard.scss` to set the margins. Feel free to merge after you do that
